src/screens/OtherProfile/elements/BioAndChat.js

```diff
@@ -0,0 +1,190 @@
+    try {
+      setLoadingSendDM(true);
+      const signedMParams = {
+        user_id: dataMain.user_id,
+        message: dmChat
+      };
+      const response = await sendSignedDMOtherProfile(signedMParams);
+      const newResponse = {...response, members: response?.message?.members};
+      await saveChatFromOtherProfile(newResponse, 'sent', true, SIGNED);
+      setDmChat('');
+    } catch (error) {
+      if (__DEV__) {
+        console.log(error, 'error');
+      }
+    } finally {
+      setLoadingSendDM(false);
+    }
+  };
+
+  const sentAnonDM = async () => {
+    try {
+      setLoadingSendDM(true);
+      const {
+        anon_user_info_emoji_name,
+        anon_user_info_emoji_code,
+        anon_user_info_color_name,
+        anon_user_info_color_code
+      } = anonProfile ?? {};
+
+      const anonDMParams = {
+        user_id: dataMain.user_id,
+        message: dmChat,
+        anon_user_info_emoji_name,
+        anon_user_info_emoji_code,
+        anon_user_info_color_name,
+        anon_user_info_color_code
+      };
+      const response = await sendAnonymousDMOtherProfile(anonDMParams);
+      await saveChatFromOtherProfile(response, 'sent', true, ANON_PM);
+      setDmChat('');
+    } catch (e) {
+      if (e?.response?.data?.status === CHANNEL_BLOCKED) {
+        const response = e?.response?.data?.data;
+        await savePendingChatFromOtherProfile(response, true);
+        setDmChat('');
+        return;
+      }
+    } finally {
+      setLoadingSendDM(false);
+    }
+  };
+
+  const onSendDM = async () => {
+    if (isAnonimity) {
+      await sentAnonDM();
+    } else {
+      await sendSignedDM();
+    }
+  };
+
+  return (
+    <View style={styles.bioAndSendChatContainer(isAnonimity)}>
+      <View style={styles.containerBio}>
+        {bio === null || bio === undefined ? (
+          <Text style={styles.bioText(isAnonimity)}>Send a message</Text>
+        ) : (
+          <Pressable onPress={openBio}>
+            <Text linkStyle={styles.seeMore} style={styles.bioText(isAnonimity)}>
+              {trimString(bio, 121)}{' '}
+              {bio.length > 121 ? <Text style={{color: colors.blue}}>see more</Text> : null}
+            </Text>
+          </Pressable>
+        )}
+      </View>
+      <TouchableOpacity
+        disabled={isSignedMessageEnabled}
+        activeOpacity={1}
+        onPress={showSignedMessageDisableToast}>
+        <TextAreaChat
+          isAnonimity={isAnonimity}
+          loadingAnonUser={loadingGenerateAnon}
+          avatarUrl={avatarUrl}
+          anonUser={anonProfile}
+          placeholder="Send a direct message"
+          disabledInput={!isSignedMessageEnabled}
+          onSend={onSendDM}
+          onChangeMessage={setDmChat}
+          disabledButton={loadingSendDM || !isSignedMessageEnabled || loadingGenerateAnon}
+          defaultValue={
+            isSignedMessageEnabled ? dmChat : `Only users ${username} follows can send messages`
+          }
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `defaultValue` prop in the `TextAreaChat` component is being set conditionally. This could lead to unexpected behavior if the conditions change after the component has been mounted, as the `defaultValue` prop does not update after initial render. Consider using a state variable instead.

```diff
- defaultValue={
-   isSignedMessageEnabled ? dmChat : `Only users ${username} follows can send messages`
- }
+ value={dmChat}
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->